### ISSUE TITLE
Are win32 DLL's of current version (2.0, or the last 1.9) are compatible with Watcom 11?

### ISSUE THREAD

**Issue by lollisoft** [open-watcom/open-watcom-v2#1352]
Hi,

I am trying to build a CORBA library, either MICO or omniORB. But both have their compiler issues with Watcom 11.

Does the compiler still produce DLL's that I can mix with Watcom 11? And try to build CORBA with OW then?

The MICO code uses some template features, that Watcom 11 does not support.
The omniORB crashes the compiler after some modifications I made to get through.

On MinGW, I can use MICO and if I remember it was able to speak to an MS VC6 build example on my remote machine, but I rely on Watcom for that part of my project using Watcom within Power++.

Thanks, Lothar

**Comment by jmalak** [comment]
What you mean by compatible with Watcom11?
It depend how you create DLL. From system point of view any DLL is not dependent on compiler, but what can be a problem if you need compatibility on entry names etc then it depends on your code and compiler/linker setup.
If you gve me more details about your problem then I could help you with compiler/linker setup.
Generaly Open Watcom is compatible with Watcom on source level not on binary level. You should be able compile source used with Watcom 11 by Open Watcom. But take into account that Open Watcom is more C standard compliant that some construct can generate Warnings etc.

**Comment by lollisoft** [comment]
Hi, I just want to integrate a CORBA code base. My code works with Watcom11 (not CORBA), but my client server code is home grown 25 years ago. I try to get a integration from Watcom in the view of a client server setup done.

Actually - just now, I tried the latest OW release snapshot with MICO. I have setup a basic project to try to build the orb. When I ignore imr and ir code for now, the next thing is, that the compiler crashes at the very next file. Normally the code compiles with MS VC 6 what I have tested.

My options are at the end of trials now, because I hoped to get MICO compiling with OW2. BTW, my earlyer issues with Watcom11 and templates are gone - so far, good. I try to port away from Power++ code. That is the reason for my trials and want to go distributed...

Maybe I could provide an up to date github code base with the mico.wpj and probably setting up INCLUDE for the mico path. But I will do that as soon as possible and not today...

**Comment by jmalak** [comment]
OK.
OW 2.0 compilers have fixed many old problems that it could compile your code without crash.
Anyway compiler should not crash if it will happen with OW 2.0 then please give here sample code to be able fix it.

**Comment by lollisoft** [comment]
Yes, I try to isolate the code and provide it here as soon as possible.

Thanks

**Comment by lollisoft** [comment]
Yes, I try to isolate the code and provide it here as soon as possible.

Thanks

**Comment by rgayoso** [comment]
I also have an old Power++ project inside a stand alone dll, loaded using LoadLibrary(). The Power++ libraries are built with Watcom 11.0b (I don't have the source code). If I build main exe with OpenWatcom, Power++ crashes when I exit exe. I think that there are some global variables shared from the dll and exe related to memory allocation. So on this project I must use 11.0b.

**Comment by jmalak** [comment]
If you put here your sample then we can analyse what happen.
Take into account that Power++ has some stuff included into Open Watcom and only compiled not tested.
Probably it is somehow incompatible with Power++.
Also there is special version of IDE configuration for Power++ not included by default.

Open Watcom Power++ stuff is in **`bld/pbide`**
![image](https://github.com/user-attachments/assets/8f380c6c-7630-4611-b2b2-fb7da4d9b2c0)
It is only build but not distributed. It should be probably install to Power++.

**Comment by lollisoft** [comment]
> I also have an old Power++ project inside a stand alone dll, loaded using LoadLibrary(). The Power++ libraries are built with Watcom 11.0b (I don't have the source code). If I build main exe with OpenWatcom, Power++ crashes when I exit exe. I think that there are some global variables shared from the dll and exe related to memory allocation. So on this project I must use 11.0b.

Hi,

I just want to ask you about some WGridComboBox issues. Did you had used that within the WGrid control at that time? I try to implement a editable lookup cell and it crashes while doing a SetEditor...

Thanks, it is just a try

**Comment by jmalak** [comment]
We don't test Open Watcom with Sybase Power++, that I cannot say anything about it.
Generaly these Watcom Classes is not used in Open Watcom code (only defined) that I cannot confirm current status.
